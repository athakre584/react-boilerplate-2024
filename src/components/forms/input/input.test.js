import { render, screen } from '@testing-library/react';
import { useForm, FormProvider } from 'react-hook-form';
import RHFInput from './input';
import React from 'react';

// Wrapper component to provide React Hook Form context
const Wrapper = ({ children }) => {
  const methods = useForm();

  return <FormProvider {...methods}>{children}</FormProvider>;
};

describe('RHFInput Component', () => {
  it('renders correctly and matches the snapshot', () => {
    // Render the component within the FormProvider
    const { asFragment } = render(
      <Wrapper>
        <RHFInput
          label="Test Label"
          fieldName="testField"
          helperText="Enter your text"
          hasParentError={false}
        />
      </Wrapper>
    );

    // Take a snapshot of the component
    expect(asFragment()).toMatchSnapshot();
  });

  it('shows error message when there is an error', () => {
    const { getByLabelText } = render(
      <Wrapper>
        <RHFInput
          label="Test Label"
          fieldName="testField"
          helperText="Enter your text"
          hasParentError={true}
        />
      </Wrapper>
    );

    // Trigger form validation and check for error message
    const inputElement = getByLabelText(/Test Label/i);
    inputElement.focus();
    inputElement.blur();

    expect(screen.getByText(/Fill out to continue/i)).toBeInTheDocument();
  });
});
