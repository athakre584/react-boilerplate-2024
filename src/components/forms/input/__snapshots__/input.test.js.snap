// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RHFInput Component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-invalid="false"
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root formField input css-y4n9bg-MuiFormControl-root-MuiTextField-root"
    data-is="TextInput"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1q964xs-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
    >
      Test Label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1o0kqxp-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-describedby=":r0:-helper-text"
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1pbc52w"
        >
          <span>
            Test Label
          </span>
        </legend>
      </fieldset>
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-wq8qsl-MuiFormHelperText-root"
      id=":r0:-helper-text"
    >
      Enter your text
    </p>
  </div>
</DocumentFragment>
`;
